source/h2_stream.c

```diff
@@ -430,6 +430,32 @@ int aws_h2_stream_on_decoder_headers_end(
     return AWS_OP_SUCCESS;
 }
 
+int aws_h2_stream_on_decoder_data(struct aws_h2_stream *stream, struct aws_byte_cursor data) {
+    AWS_PRECONDITION_ON_CHANNEL_THREAD(stream);
+
+    if (s_check_state_allows_frame_type(stream, AWS_H2_FRAME_T_DATA)) {
+        return s_send_rst_and_close_stream(stream, aws_last_error());
+    }
+
+    if (!stream->thread_data.received_main_headers) {
+        /* #TODO Not 100% sure whether this is Stream Error or Connection Error. */
+        AWS_H2_STREAM_LOG(ERROR, stream, "Malformed message, received DATA before main HEADERS");
+        return s_send_rst_and_close_stream(stream, AWS_ERROR_HTTP_PROTOCOL_ERROR);
+    }
+
+    /* #TODO Update connection's flow-control window */
```

Comment:
you have this comment in two places.... you probably don't want to handle twice tho.

Reply:
oh yeah, one of those should be "Update **stream's** flow-control window"